src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,186 @@
+     *
+     * @param filePath The relative path of the file.
+     */
+    public Storage(String filePath) {
+        this.filePath = filePath;
+        this.taskFile = new File(filePath);
+    }
+
+    /**
+     * Checks if the required directory and file is present.
+     * Creates the file and/or the directory if it is
+     * not present.
+     *
+     * @throws DukeException If the file or directory cannot
+     *                       be created.
+     */
+    public void checkFile() throws DukeException {
+        if (!taskFile.exists()) {
+            File dir = new File("data");
+            if (!(dir.exists() && dir.isDirectory())) {
+                dir.mkdir();
+            }
+            try {
+                taskFile.createNewFile();
+            } catch (IOException e) {
+                throw new DukeException("");
+
+            } catch (SecurityException e) {
+
+            }
+        }
+    }
+
+    /**
+     * Loads the current file content and adds the tasks
+     * into the TaskList if any.
+     *
+     * @return An ArrayList containing the tasks.
+     * @throws DukeException If the file cannot be found.
+     */
+    public ArrayList<Task> load() throws DukeException {
+
+        ArrayList<Task> userInput = new ArrayList<>();
+        try {
+            checkFile();
+            Scanner s = new Scanner(taskFile);
+            while (s.hasNextLine()) {
+                String taskString = s.nextLine();
+                String[] splitString = taskString.split(" \\| ");
+
+                switch (splitString[0].trim()) {
+
+                case "T":
+                    userInput.add(new Todo(splitString[2]));
+                    break;
+                case "D":
+                    DateTimeFormatter df = DateTimeFormatter.ofPattern("MMM d yyyy");
+                    LocalDate date = LocalDate.parse(splitString[3].trim(), df);
+                    userInput.add(new Deadline(splitString[2].trim(), date));
+                    break;
+                case "E":
+                    DateTimeFormatter dtf = DateTimeFormatter.ofPattern("MMM d yyyy HH:mm");
+                    LocalDateTime dateTime = LocalDateTime.parse(splitString[3].trim(), dtf);
+                    userInput.add(new Event(splitString[2], dateTime));
+                    break;
+
+                }
+
+                if (splitString.length > 2 && splitString[1].equals("Y")) {
+                    userInput.get(userInput.size() - 1).markAsDone();
+                }
+            }
+            return userInput;
+
+        } catch (FileNotFoundException e) {
+            throw new DukeException("");
+        }
+    }
+
+    /**
+     * Returns the string containing the task in
+     * a format which can be saved in the file.
+     *
+     * @param task The task to be saved in the file.
+     * @return The string representing the task.
+     */
+    public String fileString(Task task) {
+        String toAdd = task.taskIndicator() + " | " +
```

Comment:
Break line before operators (`+`) would be more in accordance to coding standard.